class-plugin-autoupdate-filter.php

```diff
@@ -10,23 +10,40 @@
 }
 
 class Plugin_Autoupdate_Filter {
+	/**
+	 * The settings loaded from the JSON file.
+	 *
+	 * @var array
+	 */
+	private $settings;
 
 	/**
-	 * Initialize WordPress hooks
+	 * Initialize WordPress hooks and load settings from JSON.
 	 */
 	public function init() {
+		$this->load_settings_from_json();
```

Comment:
We need some sort of logic to stop the plugin from initializing here if loading the settings fails. Not likely with a local file, but definitely a bigger issue if we load them over HTTP later.

Reply:
I haven't touched this one yet, because this seems like a design decision. If we can't reach the API for some reason, what would we want the default autoupdate behavior to be? 

Maybe we would want all autoupdates to be paused until the API could be reached?

Reply:
> If we can't reach the API for some reason, what would we want the default autoupdate behavior to be?

I'd pause everything. If the API is down, we'll notice and get it fixed soon. Worst case scenario, auto-updates don't happen for a bit, and I think that's better than an update going through when it shouldn't have because the server was over-loaded or something...